### PR TITLE
handle complexity endpoint for filter

### PR DESCRIPTION
**Pull request by ghazwarhili**

### CODE REVIEW

File: src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java [hunk 1]
```diff
@@ -969,6 +969,25 @@ public void testExportFilters() throws Exception {
...
+        Map<String, List<Integer>> filtersComplexityCount = objectMapper.readValue(
+                mvc.perform(post("/" + FilterApi.API_VERSION + "/filters/count")
+                                .params(params)
+                                .content(json)
+                                .contentType(APPLICATION_JSON))
+                        .andExpect(status().isOk())
+                        .andReturn().getResponse().getContentAsString(),
+                new TypeReference<>() {
+                });
+        assertEquals(3, filtersComplexityCount.size());
```
Comment (FranckLecuyer) [inline src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java:988]: I don't understand the test : 
Where are built the filters with the uuids c88c9510-15b4-468d-89c3-c1c6277966c3, 1110b06b-9b81-4d31-ac21-450628cd34ff, like in the other unit test methods ...

And contingencies are not filters ...

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/filter/server/FilterService.java [hunk 2]
```diff
@@ -695,6 +695,17 @@ public Optional<List<IdentifiableAttributes>> exportFilter(UUID id, UUID network
         return getFilter(id).map(filter -> getIdentifiableAttributes(filter, networkUuid, variantId));
     }
 
+    public Map<String, List<Integer>> containersListCount(Map<String, List<UUID>> filtersIdsMap, UUID networkUuid, String variantId) {
+        Objects.requireNonNull(filtersIdsMap);
+        Map<String, List<Integer>> resultMap = new HashMap<>();
+        for (String key : filtersIdsMap.keySet()) {
+            List<UUID> uuids = filtersIdsMap.get(key);
+            List<FilterEquipments> filtersEquipments = exportFilters(uuids, networkUuid, variantId);
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/filter/server/FilterService.java:703]: Do not use `exportFilters`

    public Map<String, List<Integer>> getIdentifiablesCount(Map<String, List<UUID>> ids, UUID networkUuid, String variantId) {
        Objects.requireNonNull(ids);
        return ids.entrySet().stream()
                .map(entry -> Map.entry(entry.getKey(), getFilters(entry.getValue()).stream().map(f -> getIdentifiableAttributes(f, networkUuid, variantId).size()).toList()))
                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
    }

File: src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java [hunk 3]
```diff
@@ -969,6 +969,66 @@ public void testExportFilters() throws Exception {
...
+        LineFilter lineFilter3 = LineFilter.builder().equipmentID("NHV1_NHV2_1").substationName1("P1").substationName2("P2")
+                .countries1(new TreeSet<>(Set.of("FR"))).countries2(new TreeSet<>(Set.of("FR")))
+                .nominalVoltage1(new NumericalFilter(RangeType.RANGE, 360., 400.)).nominalVoltage2(new NumericalFilter(RangeType.RANGE, 356.25, 393.75)).build();
+        CriteriaFilter lineCriteriaFilter3 = new CriteriaFilter(
+                filterId3,
+                new Date(),
+                lineFilter3
+        );
+        insertFilter(filterId3, lineCriteriaFilter3);
+        checkFormFilter(filterId3, lineCriteriaFilter3);
```
Comment (FranckLecuyer) [inline src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java:1009]: Better to differentiate the three filters (with distinct type, distinct equipmentId ...)
And add test on a filter with an equipment not found to test that the filterComplexityCount list size is empty in that case ...

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java (hunk 1 above)
Comment (ghazwarhili) [reply src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java]: done

File: src/main/java/org/gridsuite/filter/server/FilterService.java [hunk 4]
```diff
@@ -695,6 +695,17 @@ public Optional<List<IdentifiableAttributes>> exportFilter(UUID id, UUID network
         return getFilter(id).map(filter -> getIdentifiableAttributes(filter, networkUuid, variantId));
     }
 
+    public Map<String, List<Integer>> containersListCount(Map<String, List<UUID>> filtersIdsMap, UUID networkUuid, String variantId) {
```
[the remark this answers is not in the document]
Comment (ghazwarhili) [reply src/main/java/org/gridsuite/filter/server/FilterService.java]: done

File: src/main/java/org/gridsuite/filter/server/FilterService.java (hunk 2 above)
Comment (ghazwarhili) [reply src/main/java/org/gridsuite/filter/server/FilterService.java]: done

File: src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java (hunk 3 above)
Comment (ghazwarhili) [reply src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java]: done

**Review by SlimaneAmar — APPROVED** [review]

File: src/main/java/org/gridsuite/filter/server/FilterController.java [hunk 5]
```diff
@@ -146,6 +147,18 @@ public ResponseEntity<List<IdentifiableAttributes>> exportFilter(@PathVariable("
             .orElse(ResponseEntity.notFound().build());
     }
 
+    @PostMapping(value = "/filters/identifiables-count", produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/filter/server/FilterController.java:150]: `GetMapping` !

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

**Review by FranckLecuyer — APPROVED** [review]

**Review by FranckLecuyer — APPROVED** [review]

**Review by AAJELLAL — APPROVED** [review]

**Review by SlimaneAmar — APPROVED** [review]
